### PR TITLE
set awsclient start condition

### PR DESCRIPTION
**Pull request by motto-phytec**
The awsclient should be start after the acception of the contract and the successful onboarding of the device

### CODE REVIEW

File: recipes-connectivity/awsclient/awsclient/awsclientcheck.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+FILE="config/esec.config"
+
+if [ -f "${FILE}" ]; then
+    condition=$(jq .awsclient < $FILE | sed -r 's/["]+//g')
+    [ "$condition" = start ] && awsclient
+fi
+exit 0
```
Comment (mschwan-phytec) [inline recipes-connectivity/awsclient/awsclient/awsclientcheck.sh]: This is not ideal, as this hides the exit code of the awsclient.

File: recipes-connectivity/awsclient/awsclient/awsclientcheck.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+FILE="config/esec.config"
+
+if [ -f "${FILE}" ]; then
+    condition=$(jq .awsclient < $FILE | sed -r 's/["]+//g')
+    [ "$condition" = start ] && awsclient
```
Comment (mschwan-phytec) [inline recipes-connectivity/awsclient/awsclient/awsclientcheck.sh]: You could condense these two lines into one and insert that command directly into the service file. This way we do not need yet another shell script.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

File: recipes-connectivity/awsclient/awsclient/awsclientcheck.sh (hunk 1 above)
Comment (motto-phytec) [reply recipes-connectivity/awsclient/awsclient/awsclientcheck.sh]: is changed

File: recipes-connectivity/awsclient/awsclient/awsclientcheck.sh (hunk 2 above)
Comment (motto-phytec) [reply recipes-connectivity/awsclient/awsclient/awsclientcheck.sh]: is changed

File: recipes-connectivity/awsclient/awsclient/awsclient.service [hunk 3]
```diff
@@ -5,7 +5,7 @@ After=network.target
 [Service]
 Type=oneshot
 User=root
-ExecStart=/usr/bin/awsclient
+ExecStart=/bin/sh -c '[ $(/usr/bin/jq .awsclient < /mnt/config/aws/config/esec.config | grep start | wc -l) -eq 1 ] && /usr/bin/awsclient || exit 0'
```
Comment (mschwan-phytec) [inline recipes-connectivity/awsclient/awsclient/awsclient.service]: I think you should remove the `|| exit 0` part, as this only masks any possible errors occurring from awsclient. We want to allow for errors to stop the service.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

File: recipes-connectivity/awsclient/awsclient/awsclient.service (hunk 3 above)
Comment (motto-phytec) [reply recipes-connectivity/awsclient/awsclient/awsclient.service]: the exit 0 is for the cases

1. file /mnt/config/aws/config/esec.config not exist. the state before onboarding
2. the state is stop, it is the initial state. The customer has not yet accepted the contract.
the awsclient is in waiting state.

Comment (mschwan-phytec) [reply recipes-connectivity/awsclient/awsclient/awsclient.service]: 1. In this case the service should deliberately fail. Either exit with a new error code and echo an error message or simply let the command fail without any further error handling.
2. For this we really should disable the service by default and activate it after a successful onboarding and accepted contract. The check you are introducing with this PR can still be integrated to be sure the contract is not declined at a later point in time.

File: recipes-connectivity/awsclient/awsclient/awsclient.service [hunk 4]
```diff
@@ -5,7 +5,7 @@ After=network.target
 [Service]
 Type=oneshot
 User=root
-ExecStart=/usr/bin/awsclient
+ExecStart=/bin/sh -c '[ $(/usr/bin/jq .awsclient < /mnt/config/aws/config/esec.config | grep start | wc -l) -eq 1 ] && /usr/bin/awsclient || { echo "The contract is not accepted"; exit 2; }'
```
Comment (mschwan-phytec) [inline recipes-connectivity/awsclient/awsclient/awsclient.service]: s/The contract is not accepted/Contract has not been accepted yet/

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]
Except for some grammar fixes, this looks to be OK.

**Review by mschwan-phytec — APPROVED** [review]

**Pull request merged**